ring-core/test/ring/middleware/test/file.clj

```diff
@@ -93,3 +93,52 @@
       (is (= 200 status))
       (is (= (into #{} (keys headers)) #{"Content-Length" "Last-Modified"}))
       (is (= foo-html body)))))
+
+(deftest test-wrap-file-with-prefer-handler-option
+  (let [processable-file-uri (str "/" (.getName (File. ^String public-dir "bars.txt")))
+        handler (fn handler
```

Comment:
As I mentioned in my previous comment, I think this would be better taken out into a `defn`.

Reply:
That handler has no use outside of the test so I limited its scope to the test. it also depends on the processable-file-uri. I can hard code it but I think it's good practice for dependencies be explicit. I extracted a create-handler function to capture that dependency. Is this acceptable? if not I can hard code the processable-file-uri and have the handler as a top-level defn

Reply:
Sure, but it's unnecessary, since you're only ever calling the function with one value. There's no need to overcomplicate things for reuse that never happens.

Instead:

```clojure
(defn- prefer-foo-handler
  ([request]
   (if (= (:uri request) "/foo.html")
     {:status 200, :headers {}, :body "override"}
     {:status 404, :headers {}, :body "not found"}))
  ([request respond raise]
   (respond (prefer-foo-handler request)))))

(deftest test-wrap-file-with-prefer-handler
  (let [handler (wrap-file prefer-foo-handler
                           (File. public-dir)
                           {:prefer-handler? true})]

    (testing "middleware serves file (synchronously)"
      (let [response (handler {:request-method :get, :uri "/index.html"})]
        (is (= 200 (:status response)))
        (is (= index-html (:body response)))))

    (testing "middleware serves file (asynchronously)"
      (let [response (promise)
            error    (promise)]
        (handler {:request-method :get, :uri "/index.html"} response error)
        (is (= 200 (:status @response)))
        (is (= index-html (:body @response)))
        (is (not (realized? error)))))

    (testing "handler serves file (synchronously)"
      (let [response (handler {:request-method :get, :uri "/foo.html")]
        (is (= 200 (:status response)))
        (is (= "override" (:body response)))))
      
    (testing "handler serves file (asynchronously)"
      (let [response (promise)
            error    (promise)]
        (handler {:request-method :get, :uri "/foo.html"} response error)
        (is (= 200 (:status @response)))
        (is (= "override" (:body @response)))
        (is (not (realized? error)))))))
```

Reply:
I don't like hard-coding the url because because I want the URL to be constructed from a file. I still think the handler is better inlined in the let, but I've copied and pasted your changes. let me know if there's anything else

Reply:
Let bindings should be added to aid readability and avoid repetition. In this case we're only using `"/foo.html"` twice, and `processable-file-uri` is no more descriptive than using the literal string, so I'd prefer if you ordered the code in the way I outlined above.

Reply:
yes I incorporated your suggestion. i just forgot to squish it. its now squashed into one  commit

Reply:
There is a difference between using a string url and constructing the string from a file. The dependency is explicit. Looking "/foo.html" I would not know if that is a file that exists on the file system or an arbitrary string. In any case, this is not an important enough issue for me to fight for so I acquiesce to your decision

Reply:
Thanks. Give me a nudge when you've made the changes to the test and the commit message.

Reply:
nudge. please take a look

Reply:
Thanks for the update, but you still need to replace the test code. The reason I wrote the code in my earlier comment was to demonstrate how the test should be written out.

You also need to ensure that the commit message adheres to the [seven rules](https://chris.beams.io/posts/git-commit/#seven-rules), so ensure that it's correctly capitalized. Perhaps:

```
Add :prefer-handler? to wrap-file options

An option to reverse the default priority and check the handler before
the filesystem.
```

Reply:
OK i copied and pasted your code and commit messages exactly. take a look now. if there's anything missing, feel free to change the wording yourself. no point in me copy and pasting